server/app/services/ProgramBlockValidationFactory.java

```diff
@@ -6,6 +6,12 @@
 import models.Version;
 import repository.VersionRepository;
 
+/**
+ * Helper class for creating a ProgramBlockValidation Object.The reason this factory class is needed
+ * because - for every question in the question picker, we will need to check if the question is
+ * tombstoned, resulting in n+1 DB queries. To avoid it, we create a factory object injecting a
+ * VersionRepository and calling getDraftVersion() only once during the object creation.
```

Comment:
**nitpick:** 
```suggestion
 * Factory for creating `ProgramBlockValidation` instances. 
 * <p>Needed because - for every question in the question picker, we will need to check if the question is
 * tombstoned, resulting in n+1 DB queries. To avoid this, we create a factory object injecting a
 * VersionRepository and calling getDraftVersion() only once per request.
```